src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+package parser;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import commands.AddCommand;
+import commands.Command;
+import commands.DateTimeCommand;
+import commands.DeleteCommand;
+import commands.ExitCommand;
+import commands.FindCommand;
+import commands.ListCommand;
+import commands.MarkCommand;
+import javafx.scene.layout.VBox;
+import storage.Storage;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+import tasklist.TaskList;
+import ui.Ui;
+
+public class Parser {
+
+	enum TaskType {
+		TODO, DEADLINE, EVENT
+	}
+
+	/**
+	 * Get command from input.
+	 * 
+	 * @param input
+	 * @return Command
+	 */
+	public Command parseInput(String input) {
+		String[] inputArr = input.split(" ");
+		String opType = inputArr[0].toLowerCase();
+		switch (opType) {
+			case "bye":
+				return new ExitCommand(inputArr);
+			case "list":
+				return new ListCommand(inputArr);
+			case "datetime":
+				return new DateTimeCommand(inputArr);
+			case "mark":
+			case "unmark":
+				return new MarkCommand(inputArr, opType.startsWith("mark"));
+			case "delete":
+				return new DeleteCommand(inputArr);
+			case "todo":
+			case "deadline":
+			case "event":
+				return new AddCommand(inputArr);
+			case "find":
+				return new FindCommand(inputArr);
+			default:
+				return new Command(inputArr);
+		}
+	}
+
+	/**
+	 * Create task from record in saved file and add it to tasklist.
+	 * 
+	 * @param input
+	 * @param tasklist
+	 * @param ui
+	 * @param storage
+	 * @param dialogContainer
+	 */
+	public void parseAndSetData(String input, TaskList tasklist, Ui ui, Storage storage, VBox dialogContainer) {
+		String inputLower = input.toLowerCase();
+		if (inputLower.isBlank()) {
+			return;
+		}
+		String[] inputArr = inputLower.split("\\|");
+		TaskType operationType = inputLower.startsWith("t")
+				? TaskType.TODO
+				: inputLower.startsWith("d")
+						? TaskType.DEADLINE
+						: TaskType.EVENT;
+
+		Task newTask = null;
+
+		switch (operationType) {
+			case TODO:
+				newTask = new Todo(inputArr[2]);
+				break;
+
+			case DEADLINE:
+				newTask = new Deadline(inputArr[2], parseDateTimeStr(inputArr[3], ui, storage, dialogContainer));
+				break;
+
+			case EVENT:
+				newTask = new Event(inputArr[2],
+						new LocalDateTime[] { parseDateTimeStr(inputArr[3], ui, storage, dialogContainer),
+								parseDateTimeStr(inputArr[4], ui, storage, dialogContainer) });
+				break;
+
+			default:
+				break;
+		}
+		if (inputArr[1].equals("1"))
+			newTask.markAsDone();

```

Comment:
To fit the coding standards, you could use opening and closing braces for this `if` condition